app/lib/actions/helpers/smart_proxy_sync_history_helper.rb

```diff
@@ -0,0 +1,17 @@
+module Actions
+  module Helpers
+    module SmartProxySyncHistoryHelper
+      def self.included(base)
+        base.middleware.use ::Actions::Middleware::RecordSmartProxySyncHistory
+      end
+
+      def done?
+        is_done = super
+        if is_done
+          ::Katello::SmartProxySyncHistory.where(:id => output[:smart_proxy_history_id]).update_all(finished_at: Time.now)
+        end
```

Comment:
this still needs to check for errors, right?

Reply:
Leaning towards leaving the record as is. In case, there's failures, we will have a sync history record without finished_at which will be cleared upon the next sync..

Reply:
but if there are errors, it won't ever sync again, right?
Say i sync my some my capsule and it errors, so i sync it again.  If finished_at is set, it won't sync again?

Reply:
and when i say 'it won't ever sync again' i mean until there are changes to the repo

Reply:
oh and one little nitpick, you might change it to:
```
           ::Katello::SmartProxySyncHistory.where(:id => output[:smart_proxy_history_id], :finished_at => nil).update_all(finished_at: Time.now)
```
So it won't re-update anything if done? is called multiple times after its completed

Reply:
> and when i say 'it won't ever sync again' i mean until there are changes to the repo

So it will pick up all repos without a sync history record as well as records without the finished_at set for the next sync.